robottelo/ui/role.py

```diff
@@ -85,3 +85,29 @@ def add_permission(self, role_name, resource_type=None,
             self.click(tab_locators['roles.tab_location'])
             self.configure_entity(location, FILTER['filter_loc'])
         self.click(common_locators['submit'])
+
+    def get_resources(self, role_name):
```

Comment:
`>10` condition sounds really bad for me. `get_permissions` seems as good enough method, but `get_resources` has to be re-implemented using that internal search/filter. So we can pass a list here and input each resource one by one to check whether it is really present in the system and read permissions if necessary